### ISSUE TITLE
Processes started with bash file `flutter/bin/dart` do not terminate children processes when killed

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/flutter#98435]
`which dart` on my machine points to `path/to/flutter/bin/dart`. I was writing a test in DDS and attempting to start a process:
```
final process = await Process.start('dart', args);
```
Since `dart` points to `flutter/bin/dart`, this actually starts a bash process, not a dart process. Later in my code, I attempted to kill this process: `process.kill();` and was checking that other parts of the ecosystem responded accordingly to the process being killed (specifically, I was verifying that DevTools would disconnect). Calling `process.kill()` killed the bash process, but not any subprocesses that were started by the bash process (i.e. the dart process that DevTools should be responding to). I expect any user who points their Dart to `flutter/bin/dart` and is trying to spawn a process will hit this issue. 

@bkonyi and I spent time debugging this this morning. Please add any additional details or thoughts.

**Comment by maheshmnj** [comment]
Hi @kenzieschmoll, Thanks for filing the issue. Would it be possible to share a minimal reproducible code? 
This looks related to https://github.com/flutter/flutter/issues/98435

**Comment by christopherfujino** [comment]
It looks like this is blocked by https://github.com/dart-lang/sdk/issues/22470

**Comment by bkonyi** [comment]
Totally thought I followed up on this, but I guess I didn't. This isn't related to dart-lang/sdk#22470 but is an issue with the wrapper scripts we have around the standalone Dart VM in the Flutter SDK. We should be `exec`ing the VM from within those wrappers, but instead we're spawning a child process. It's a rather simple fix IIRC.

**Comment by christopherfujino** [comment]
> Totally thought I followed up on this, but I guess I didn't. This isn't related to [dart-lang/sdk#22470](https://github.com/dart-lang/sdk/issues/22470) but is an issue with the wrapper scripts we have around the standalone Dart VM in the Flutter SDK. We should be `exec`ing the VM from within those wrappers, but instead we're spawning a child process. It's a rather simple fix IIRC.

Ahh cool

**Comment by christopherfujino** [comment]
@gspencergoog any reason not to call `exec` from our bash entrypoints rather than spawning a subprocess for flutter and dart?

**Comment by gspencergoog** [comment]
No, no reason at all.  That's the right thing to do.

**Comment by darshankawar** [comment]
Labeling it as applicable based on the discussion / thread above.

**Comment by christopherfujino** [comment]
> No, no reason at all. That's the right thing to do.

cool, I can open a PR for that

**Comment by slightfoot** [comment]
I can confirm this is still an issue on windows with the batch scripts used.

**Comment by maheshmnj** [comment]
@slightfoot Could you please file that as a separate issue?  since this thread has been locked and limited to collaborators only.

**Comment by christopherfujino** [comment]
> I can confirm this is still an issue on windows with the batch scripts used.

I agree with @maheshmnj, please file a new issue for that; this issue was specifically for bash entrypoint.
